src/main/java/com/fasterxml/jackson/databind/DefaultCacheProvider.java

```diff
@@ -0,0 +1,75 @@
+package com.fasterxml.jackson.databind;
+
+import com.fasterxml.jackson.databind.deser.DeserializerCache;
+import com.fasterxml.jackson.databind.util.LookupCache;
+
+import java.util.function.Supplier;
+
+/**
+ * Default implementation of {@link CacheProvider} that provides easy, builder-based custom cache configuration
+ * using {@link DefaultCacheProvider.Builder}.
+ * <p>
+ * Users can either use this class or implement their own {@link CacheProvider} imlementation.
+ * 
+ * @since 2.16
+ */
+public class DefaultCacheProvider
+    implements CacheProvider
+{
+
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * Cache instance to be used by {@link DeserializerCache}.
+     */
+    protected final LookupCache<JavaType, JsonDeserializer<Object>> _deserializerCacheSupplier;
```

Comment:
I am not sure need or want supplier, but overridable method that will construct `LookupCache<T>` for any given type `T`: and the default implementation just constructs `LRUMap<T>`?

But if we do want supplier it should be generic, used for all caches.